### PR TITLE
Regression: Unavailable devices in unsupported browsers

### PR DESCRIPTION
**Pull request by MartinSchoeler**
This PR disables device changes in browsers that not support the `setSinkId` method, and therefore does not support device changes directly from the browser

Screen Shot 2022-07-07 at 11 18 10 AM

I've decided to test if the `setSinkId` is available instead of listing browsers that supported or not, this way if a browser starts/stops supporting it, we don't have to do anything, and there is also firefox, where you can enable it as an optional feature.

### CODE REVIEW

**Review by ggazzo — CHANGES_REQUESTED** [review]

File: apps/meteor/ee/client/voip/modals/DeviceSettingsModal.tsx [hunk 1]
```diff
@@ -23,7 +24,7 @@ const DeviceSettingsModal = (): ReactElement => {
 			outputDevice: selectedAudioDevices?.audioOutput?.id || '',
 		},
 	});
-
+	const setSinkIdAvailable = useMemo(() => isSetSinkIdAvailable(), []);
```
Comment (ggazzo) [inline apps/meteor/ee/client/voip/modals/DeviceSettingsModal.tsx:27]:
```suggestion
	const [setSinkIdAvailable] = useState(() => isSetSinkIdAvailable());
```

File: apps/meteor/packages/rocketchat-i18n/i18n/en.i18n.json [hunk 2]
```diff
@@ -1485,6 +1485,7 @@
   "Desktop_Notifications_Enabled": "Desktop Notifications are Enabled",
   "Desktop_Notifications_Not_Enabled": "Desktop Notifications are Not Enabled",
   "Details": "Details",
+  "Device_Changes_Not_Available": "Device changes not available in this browser. For garanteed availability, please use Rocket.Chat's official desktop app.",
```
Comment (KevLehman) [inline apps/meteor/packages/rocketchat-i18n/i18n/en.i18n.json:1488]:
```suggestion
  "Device_Changes_Not_Available": "Device changes not available in this browser. For guaranteed availability, please use Rocket.Chat's official desktop app.",
```

**Review by KevLehman — CHANGES_REQUESTED** [review]

**Comment by KevLehman** [comment]
🤔 since the description goes to release, can you put a note mentioning that there's "experimental" support for the feature on Firefox through flags? https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/setSinkId. Not recommended for longer use, but at least for trying out.

**Comment by MartinSchoeler** [comment]
> 🤔 since the description goes to release, can you put a note mentioning that there's "experimental" support for the feature on Firefox through flags? https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/setSinkId. Not recommended for longer use, but at least for trying out.

I wanted to avoid talking about browsers though, since today is experimental but tomorrow it might change

File: apps/meteor/ee/client/voip/modals/DeviceSettingsModal.tsx [hunk 3]
```diff
@@ -51,13 +52,23 @@ const DeviceSettingsModal = (): ReactElement => {
 				<Modal.Close onClick={onCancel} />
 			</Modal.Header>
 			<Modal.Content fontScale='p2'>
+				{!setSinkIdAvailable && (
+					<Box color='danger-600' display='flex' flexDirection='column'>
+						{t('Device_Changes_Not_Available')}
+						<Box is='a' href='https://rocket.chat/download'>
+							{t('Desktop_App_Download')}
```
Comment (KevLehman) [inline apps/meteor/ee/client/voip/modals/DeviceSettingsModal.tsx:59]:
```suggestion
							{t('Download_Destkop_App')}
```

**Review by KevLehman — APPROVED** [review]

File: apps/meteor/ee/client/voip/modals/DeviceSettingsModal.tsx [hunk 4]
```diff
@@ -51,13 +52,23 @@ const DeviceSettingsModal = (): ReactElement => {
 				<Modal.Close onClick={onCancel} />
 			</Modal.Header>
 			<Modal.Content fontScale='p2'>
+				{!setSinkIdAvailable && (
+					<Box color='danger-600' display='flex' flexDirection='column'>
+						{t('Device_Changes_Not_Available')}
+						<Box is='a' href='https://rocket.chat/download'>
```
Comment (KevLehman) [inline apps/meteor/ee/client/voip/modals/DeviceSettingsModal.tsx:58]:
```suggestion
						<Box is='a' href='https://rocket.chat/download' target='_blank' rel='noopener noreferrer'>
```

**Review by aleksandernsilva — APPROVED** [review]

**Pull request merged**